services/github/github-api-provider.js

```diff
@@ -1,6 +1,16 @@
 'use strict'
 
+const Joi = require('joi')
 const { TokenPool } = require('../../lib/token-pool')
+const { nonNegativeInteger } = require('../validators')
+
+const headerSchema = Joi.object({
+  'x-ratelimit-limit': nonNegativeInteger,
+  'x-ratelimit-remaining': nonNegativeInteger,
+  'x-ratelimit-reset': nonNegativeInteger,
+})
+  .required()
+  .unknown(true)
```

Comment:
This is another new Joi feature for me (have been reading https://github.com/hapijs/joi/blob/v14.3.1/API.md#objectunknownallow). I take it this allows for unknown keys, but does this mean that there are keys we don't know about but want to have available on the object?

Reply:
Without `.unknown(true)` this always fails because there are a lot of other headers in the response.

This is where we invoke `Joi.validate` via `BaseService#_request`:

https://github.com/badges/shields/blob/c4efdc8e66ec47250a7e90e666ff312abe226db6/lib/validate.js#L21-L24

`stripUnknown` means keys that aren't in the schema get stripped out, which is nice for the services because it means you can't accidentally use anything that hasn't been validated.

It's not clear whether we can pass that option here.

Reply:
Ahhh okay, thanks!